📖 Improve the doc content for the WDS page #3000

Tob-iee opened the pull request:
## Summary

## Related issue(s)

3000

Fixes #

- Moved introductory content ("A Workload Description Space (WDS) is ...") to the appropriate section, under "What is a WDS?"
- Updated documentation to reflect that currently, the only supported way to create a WDS is with the core Helm chart.
- Fixed rendering Indentation for the WDS creation section
- Clarified that the kubestellar-controller-manager and transport-controller run in the KubeFlex hosting cluster, not in the WDS.
- Updated the "Accessing the WDS" section to explain the distinction between the two commands for creating/updating the kubeconfig context.
- Added a link on the page to reference the architecture page.

MikeSpreitzer commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
 ## What is a WDS?
 
-A WDS is a space (a Kubernetes-like API server with storage) that:
+A Workload Description Space (WDS) is a space in the [KubeStellar architecture](docs/content/direct/user-guide-intro.md) that serves as the primary interface for users to define and manage workloads for multi-cluster deployment. The WDS constitue of a Kubernetes API server with storage that:
```
<img width="1352" alt="image" src="https://github.com/user-attachments/assets/928748a8-b3ff-4b4c-bf69-c4054dbd6ceb" />

MikeSpreitzer commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
+A Workload Description Space (WDS) is a space in the [KubeStellar architecture](docs/content/direct/user-guide-intro.md) that serves as the primary interface for users to define and manage workloads for multi-cluster deployment. The WDS constitue of a Kubernetes API server with storage that:
 
 - Stores the definitions of workloads in their native Kubernetes format
 - Hosts the control objects (`BindingPolicy` and `Binding`) that define how workloads are distributed
```
Quoting #3000:

> There are additional kinds of control objects that can appear in a WDS.

MikeSpreitzer commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
 ## Creating a WDS
 
-A WDS can be created in several ways:
+Currently, we only support the use of our Kubestellar core Helm chart as the only way to  create a WDS. This is because the core Helm chart also automatically creates a kubestellar-controller-manager and transport-controller, which are contorllers that your WDS requires to function properly.
```
Not great grammar: double use of "only" here.

Not a blocker.

MikeSpreitzer commented on docs/content/direct/wds.md:
````diff
@@ -2,68 +2,87 @@
 ```shell
 helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
-  --set-json='WDSes=[{"name":"wds1", "type":"k8s"}]'
+  --set-json='WDSes=[{"name":"<wds-name>", "type":"<space-type>"}]'
````
The variable names in angle brackets here should be the same as the variable names listed below.

MikeSpreitzer commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
+    - `k8s` (default): Creates a basic Kubernetes API Server with a subset of kube controllers
+    - `host`: Uses the KubeFlex hosting cluster itself. 
+    
 - `APIGroups`: A comma-separated list of API Groups to include
 - `ITSName`: The name of the ITS to be used by this WDS (required if multiple ITSes exist)
```
These do not appear in the exhibited command. Probably because of https://github.com/kubestellar/kubestellar/pull/3023/files#r2178389286 .

The existing text has this defect, which does not need to be fixed in this PR.

MikeSpreitzer commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
 - `ITSName`: The name of the ITS to be used by this WDS (required if multiple ITSes exist)
 
-### Using the KubeFlex CLI
+The type of space you choose determines the type of controllers that are included in your your WDS. However, when creating a WDS, the recommended space types are "K8s" (the default space) and "host". If you the decide to use your hosting cluster's control plane (which is typically a KubeFlex cluster for Kubstellar) by specifying `type: host`, it presents you some benefits such as:
```
"However" is not the right connector here, the sentence that it introduces is not in tension with the previous remark. I would simply remove that word.

MikeSpreitzer commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
+The type of space you choose determines the type of controllers that are included in your your WDS. However, when creating a WDS, the recommended space types are "K8s" (the default space) and "host". If you the decide to use your hosting cluster's control plane (which is typically a KubeFlex cluster for Kubstellar) by specifying `type: host`, it presents you some benefits such as:
 
-You can also create a WDS using the KubeFlex CLI:
+- Prevents you from creating a separate control plane
```
"Prevents you" is a bit of a misdirection. The user does not directly create ControlPlane objects, that is done indirectly through the chart that this text is talking about. I think a more on-point description would be something like one of the following.

> Avoids creating a separate space.

> Saves the cost of creating an additional space.

MikeSpreitzer commented on docs/content/direct/wds.md:
````diff
@@ -2,68 +2,87 @@
-```shell
-kflex create wds1 -p kubestellar
-```
+> You can create multiple WDSes of the same space types and even a mix of different space types, where you can have the default and host WDSes in your cluster. A multiple WDS architecure can be valueable in cases where you want to manage workload runs across different sets of users or groups. This each users or groups can use thesame ITS for workload execution but with their different WDS
````
s/This/Thus/

s/each users or groups/each user or group/

s/thesame/the same/

s/for workload execution/for WEC inventory/

MikeSpreitzer commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
+Once your WDS is created, you can:
+
+1. **Create workload objects** in their native Kubernetes format
+2. **Define BindingPolicy objects** to specify which workloads should be deployed to which WECs
```
There are also other kinds of control objects.

Tob-iee commented on docs/content/direct/wds.md:
```diff
@@ -2,68 +2,87 @@
 - Maintains status information about deployed workloads
 - Acts as the main user interface to the KubeStellar system
 
+> A space is an abstraction (or a light weight version) of the typical Kubernetes control plane that includes just the Kubernetes API server and a subset of relevant controllers. There are different types of spaces and your choice depend on your use-case requirements.
```
I just wanted to highlight and remind to the user what a "space" is, but i think i will just take the entire thing out since it's already explained in the architecture page.

Tob-iee commented:
> Please fix the reference to Issue #3000 in the opening comment.

@MikeSpreitzer I have addressed the comments you mentioned. Kindly review these new changes when you can get to it.

MikeSpreitzer commented:
@Tob-iee: there is a difference between

```
Fixes #3000
```

and

```
3000

Fixes #
```

GitHub recognizes the first as asserting that this PR resolves Issue 3000, but does not recognize the second.

MikeSpreitzer commented:
This PR overlaps with #2993. I have asked the author of that PR (@rishi-jat) to factor it into several smaller PRs. Please collaborate with him on updates to the WDS page.
